go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/sigstore/sigstore
 
-go 1.18
+go 1.20
```

Comment:
echoing @haydentherapper and this means the minimum required version to build this project

I would keep this as is for now and maybe we bump to go1.19 in a near future